cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/distributed/AbstractProgramTwillRunnable.java

```diff
@@ -454,6 +483,12 @@ public Cancellable announce(String serviceName, int port) {
 
           // For binding StreamWriter
           install(createStreamFactoryModule());
+
+          // bind AuthorizationEnforcer to AuthorizationEnforcementService
+          bind(AuthorizationEnforcer.class).to(AuthorizationEnforcementService.class).in(Scopes.SINGLETON);
+          // also bind AuthorizationEnforcementService as a singleton. This binding is used while starting/stopping
+          // the service itself.
+          bind(AuthorizationEnforcementService.class).in(Scopes.SINGLETON);
```

Comment:
It's better not to expose the implementation. Shouldn't it be the `Authorizer` to be exposed?